packages/core/src/components/Timer.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  useImperativeHandle,
+  forwardRef,
+} from "react";
+import { Text, StyleSheet, TextStyle } from "react-native";
+
+interface TimerProps {
+  style?: TextStyle;
+  initialTime?: number;
+  updateInterval?: number;
+  format?: "ss" | "mm:ss" | "hh:mm:ss" | "ss:ms" | "mm:ss:ms" | "hh:mm:ss:ms";
+  onTimerChange?: (time: number) => void;
+  onTimerEnd?: () => void;
+  countDirection?: "up" | "down";
+}
+
+export interface TimerHandle {
+  start: () => void;
+  stop: () => void;
+  reset: (newTime?: number) => void;
+}
+
+const Timer = forwardRef<TimerHandle, TimerProps>(
+  (
+    {
+      style,
+      initialTime,
+      updateInterval = 1000,
+      format = "mm:ss",
+      onTimerChange,
+      onTimerEnd,
+      countDirection = "up",
+    },
+    ref
+  ) => {
+    const defaultInitialTime = countDirection === "up" ? 0 : 100000;
+    const [time, setTime] = useState(initialTime ?? defaultInitialTime);
+    const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+    useEffect(() => {
+      onTimerChange?.(time);
+    }, [time, onTimerChange]);
```

Comment:
```suggestion
    }, [time]);
```
I would remove the callback from here. I know it will warn you otherwise, but in most cases this callback will be set as an anonymous function. Something like `onTimerChange={()=>{...}}`, which means it will cause the useEffect to be called on every screen re-render since a new anonymous function is created everytime. We had this issue with several components so far and had to change it. It won't cause any issues because the use effect will still use the updated onTimerChange everytime the `time` changes.

Add a `// eslint-disable-next-line react-hooks/exhaustive-deps` to ignore the warning.

Reply:
Thanks, good point 👍